embeddings/evaluator/evaluator.py

```diff
@@ -13,3 +15,19 @@ def __init__(self) -> None:
     @abc.abstractmethod
     def evaluate(self, data: Input) -> Output:
         pass
+
+    def persisting(self, persister: ResultsPersister[Output]) -> "Evaluator[Input, Output]":
```

Comment:
is it necessary to use quotes? quotes are designed for not defined types https://www.python.org/dev/peps/pep-0484/#forward-references

Reply:
and that is a case, we are referencing to evaluator type in the definition of it